Lib/dis.py

```diff
@@ -32,25 +32,30 @@ def _try_compile(source, name):
     return c
 
 def dis(x=None, *, file=None, depth=None):
-    """Disassemble classes, methods, functions, generators,
-       asynchronous generators, coroutines, or code.
+    """Disassemble classes, methods, functions, other compiled objects, or code.
 
-    With no argument, disassemble the last traceback.
+       With no argument, disassemble the last traceback.
 
+       Compiled objects currently include generator objects, async generator
+       objects, and coroutine objects, all of which store their code object
+       in a special attribute.
     """
     if x is None:
         distb(file=file)
         return
-    if hasattr(x, '__func__'):  # Method
+    # Extract functions from methods.
+    if hasattr(x, '__func__'):
         x = x.__func__
-    if hasattr(x, '__code__'):  # Function
+    # Extract compiled code objects from...
+    if hasattr(x, '__code__'):  # ...a function, or
         x = x.__code__
-    if hasattr(x, 'gi_code'):  # Generator
+    elif hasattr(x, 'gi_code'):  #...a generator object, or
```

Comment:
This struck me as well, so I did a little digging on the history and context. It looks as though `gi_code` was added to generators in bpo-1473257. At this time, function bytecode was still stored in `f.func_code`, so `gi_code` was a clear analogy.

Then `func_code` was changed in 3.0 to `__code__` as part of the `f.func_X` to `f.__X__` renaming. The 3.x whatsnew explained that the purpose of this change was to "free up [the `f.func_X`] names in the function attribute namespace for user-defined attributes". But this wasn't done for generators, even though they had more or less the same `gen.gi_X` objects. On a quick look, I didn't find any discussion of this at the time, but that doesn't mean there wasn't any.

It seems, then, that there are two questions now:

1. Should the code objects (etc.) in these three types also be dundered for namespace reasons;
2. Should the equivalent objects in generators, coroutines, etc. share names or is there a reason why (following bpo-1473257) they should have different names (whether or not dundered)?

1 seems easier to answer, although that kind of breaking change was probably more in the spirit of 3.0 than 3.7. I have a couple of dim thoughts on 2 but I'll save them for possible later discussion elsewhere...